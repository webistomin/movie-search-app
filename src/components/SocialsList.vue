<template>
  <ul class="socials__list"
      v-if="isListVisible">
    <socials-list-item
      v-for="review of reviewsList"
      :key="review.id"
      :review-author="review.author"
      :review-message="review.content"
    />
  </ul>
</template>

<script>
  import SocialsListItem from './SocialsListItem';

  export default {
    name: 'SocialsList',
    components: { SocialsListItem },
    props: {
      reviewsList: {
        type: Array,
      },
    },
    computed: {
      isListVisible() {
        return this.reviewsList && this.reviewsList.length !== 0 && !this.getLoadingState;
      },
      getLoadingState() {
        return this.$store.getters.getLoadingState;
      },
    },
  };
</script>

<style>

</style>
