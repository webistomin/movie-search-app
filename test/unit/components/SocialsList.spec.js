import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import SocialsList from '../../../src/components/SocialsList';
import routes from '../../../src/router/';

describe('Header.vue', () => {
  let actions;
  let store;
  let state;
  let mutations;
  let getters;
  let router;
  let vm;
  let localVue;
  let mockedRoute;

  beforeEach(() => {
    localStorage.clear();
    jest.clearAllMocks();
    jest.resetModules();

    actions = {};

    mutations = {};

    state = {
      isLoading: false,
    };

    getters = {
      getLoadingState: state => state.isLoading,
    };

    store = new Vuex.Store({
      state,
      mutations,
      actions,
      getters,
    });

    localVue = createLocalVue();
    localVue.use(Vuex);
    router = routes;
  });

  it('Matches snapshot', () => {
    vm = shallowMount(SocialsList, {
      store,
      localVue,
      router,
      propsData: {
        reviewsList: [{ id: 0, content: 'Good movie', author: 'Alexey' }],
      },
    }).vm;
    expect(vm.$el).toMatchSnapshot();
  });

  describe('test isListVisible computed', () => {
    it('return true if array isnt empty and isnt loading', () => {
      vm = shallowMount(SocialsList, {
        store,
        localVue,
        router,
        propsData: {
          reviewsList: [{ id: 0, content: 'Good movie', author: 'Alexey' }],
        },
      }).vm;
      state.loadingState = false;
      expect(vm.isListVisible).toEqual(true);
    });
  });
});
