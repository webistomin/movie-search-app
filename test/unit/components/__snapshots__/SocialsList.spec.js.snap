// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Matches snapshot 1`] = `
<ul
  class="socials__list"
>
  <socials-list-item-stub
    reviewauthor="Alexey"
    reviewmessage="Good movie"
  />
</ul>
`;
